src/profile-logic/active-tab.js

```diff
@@ -124,6 +123,11 @@ export function computeActiveTabTracks(
     }
   }
 
+  const mainTrack: ActiveTabMainTrack = {
+    type: 'tab',
+    mainThreadIndex: mainTrackIndexes[0] || 0,
```

Comment:
Optional, maybe not for this PR: When does this `|| 0` happen? Should this be an error? Should we switch back to some other view if this one fails? This smells like a failure case to me. Although, looking at it, this is the same behavior as the existing implementation, it just stands out here more. At the very least I would add a comment about it.

Thinking out loud, no action needed:  It kind of makes me wish we had user-visible warnings we could show, rather than just console.error. For example showing a small modal that says "No active thread was found for this profile, defaulting to the first thread."

Reply:
Yeah, this is something we need to fix and I was thinking about this for a while. I think I have a comment about this inside the receive-profile file. We should ideally revert to the full view with an explanation why it didn't work. I'm adding a comment here as well.